Devices/NUCLEO411RET6/DeviceSelector.h

```diff
@@ -0,0 +1,163 @@
+
+// PWM
+#define MAX_PWM_PER_CONTROLLER               4
+#define TOTAL_PWM_CONTROLLER                 4
+#define STM32F4_PWM                     {   {TIM1  ,  STM32F4_Gpio_AlternateFunction::AF1,     { _P(A, 8) , _P(A, 9) , _P(A,10) , _P_NONE_} ,  { false, false, false, false }, 0.0, 0.0, {0.0, 0.0, 0.0, 0.0}, 0, 0, 1} ,\
+                                            {TIM2  ,  STM32F4_Gpio_AlternateFunction::AF1,     { _P(A, 5) , _P(B, 3) , _P(A, 2) , _P(A, 3)} ,  { false, false, false, false }, 0.0, 0.0, {0.0, 0.0, 0.0, 0.0}, 0, 0, 2} ,\
+                                            {TIM3  ,  STM32F4_Gpio_AlternateFunction::AF2,     { _P(C, 6) , _P(B, 5) , _P(C, 8) , _P(C, 9)} ,  { false, false, false, false }, 0.0, 0.0, {0.0, 0.0, 0.0, 0.0}, 0, 0, 3} ,\
+                                            {TIM4  ,  STM32F4_Gpio_AlternateFunction::AF2,     { _P(B, 6) , _P(B, 7) , _P(B, 8) , _P(B, 9)} ,  { false, false, false, false }, 0.0, 0.0, {0.0, 0.0, 0.0, 0.0}, 0, 0, 4} }
+
+// I2C
+#define STM32F4_I2C_PORT                1
+#define STM32F4_I2C_SCL_PIN             _P(B, 8)
+#define STM32F4_I2C_SDA_PIN             _P(B, 9)
+
+// SPI
+#define TOTAL_SPI_CONTROLLERS           2
+#define STM32F4_SPI_SCLK_PINS           { _P(B, 3), _P(B,13) }
+#define STM32F4_SPI_MISO_PINS           { _P(B, 4), _P(B,14) }
+#define STM32F4_SPI_MOSI_PINS           { _P(B, 5), _P(B,15) }
+
+// UART
+#define TOTAL_UART_CONTROLLERS          2  // changed to fit STM32F411RET6 capacity
+#define STM32F4_UART_TX_BUFFER_SIZE     256
+#define STM32F4_UART_RX_BUFFER_SIZE     512
+#define STM32F4_UART_RXD_PINS           { _P(A,10), _P(A, 3) }
+#define STM32F4_UART_TXD_PINS           { _P(A, 9), _P(A, 2) }
+#define STM32F4_UART_CTS_PINS           { GPIO_PIN_NONE, GPIO_PIN_NONE }
+#define STM32F4_UART_RTS_PINS           { GPIO_PIN_NONE, GPIO_PIN_NONE }
+
+//  USBC
+#define TOTAL_USB_CONTROLLER            1
+#define USB_MAX_QUEUES                  16
+#define USB_VENDOR_ID                   0x1B9F
+#define USB_PRODUCT_ID                  0x0110
+#define USB_MANUFACTURER_NAME           {'S', 'T', 'M'}
+#define USB_PRODUCT_NAME                 {'S', 'T', 'M', '3', '2', 'F', '4', '1', '1', 'N', 'u', 'c', 'l', 'e', 'o', ' '}
+#define USB_DISPLAY_NAME                USB_PRODUCT_NAME
+#define USB_FRIENDLY_NAME               USB_PRODUCT_NAME
+
+
+// Debug
+#define DEBUG_TEXT_PORT                 USB1
+#define STDIO                           USB1
+#define DEBUGGER_PORT                   USB1
+#define MESSAGING_PORT                  USB1
+
+#define LMODE_PIN                       _P(B,10)
+#define LMODE_USB_STATE                 TinyCLR_Gpio_PinValue::High
+
+// Loader
+#define RUNAPP_PIN						_P(C,13)
+#define RUNAPP_STATE 				    TinyCLR_Gpio_PinValue::High
+
+#define UART_DEBUGGER_INDEX 1           //allow debug from SERIAL PORT
+#define USB_DEBUGGER_INDEX 0
+
+#define RAM_BOOTLOADER_HOLD_ADDRESS 0x20017FF8
```

Comment:
Since there is no GHI bootloader on this device, these two values are not needed.